# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/DrawTargetWebgl.h
# Commit: 20e06cb8a8fd
# Full Hash: 20e06cb8a8fde5b133d3efa7cca3b92f127cf027
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-03-15 21:39:24
# Description:
#   Bug 1755458 - Use threadlocal shared WebGL context for accelerated canvas. r=lsalzman
#   
#   Because ClientWebGLContext is single threaded, we need a different
#   SharedContext for each thread that an accelerated Canvas2D is created
#   for.
# ==============================================================================

diff -r c09df30d2249 -r 20e06cb8a8fd dom/canvas/DrawTargetWebgl.h
--- a/dom/canvas/DrawTargetWebgl.h	Tue Mar 15 17:35:22 2022 +0000
+++ b/dom/canvas/DrawTargetWebgl.h	Tue Mar 15 17:37:30 2022 +0000
@@ -10,6 +10,7 @@
 #include "mozilla/gfx/2D.h"
 #include "mozilla/LinkedList.h"
 #include "mozilla/WeakPtr.h"
+#include "mozilla/ThreadLocal.h"
 #include <vector>
 
 namespace mozilla {
@@ -98,7 +99,6 @@
     SharedContext();
     ~SharedContext();
 
-    int32_t mNumTargets = 0;
     WeakPtr<DrawTargetWebgl> mCurrentTarget;
     IntSize mViewportSize;
     IntRect mClipRect;
@@ -226,7 +226,7 @@
 
   RefPtr<SharedContext> mSharedContext;
 
-  static RefPtr<SharedContext> sSharedContext;
+  static MOZ_THREAD_LOCAL(SharedContext*) sSharedContext;
 
  public:
   DrawTargetWebgl();
